lib/rb-kqueue/native.rb

```diff
@@ -14,13 +14,23 @@ module Native
     # @private
     class KEvent < FFI::Struct
       if FFI::Platform::IS_FREEBSD
-        layout(
+        fields = [
           :ident,  :uintptr_t,
           :filter, :short,
           :flags,  :u_short,
           :fflags, :u_int,
           :data,   :intptr_t,
-          :udata,  :pointer)
+          :udata,  :pointer ]
+
+        # The FFI gem incorrectly determines the freebsd version, by ignoring
+        # the decimal place
+        freebsd_version = RbConfig::CONFIG['host_os'].gsub(/[^\d.]/, '').to_f
+
+        if freebsd_version >= 12.0
+          fields.push(:ext, [ :u_int64_t, 4 ])
```

Comment:
This is the only line I don't fully understand. While I'm a maintainer, I'm not the original author so I'm a little more conservative in what I merge and release; I want to verify that `[:u_int64_t, 4]` is correct here and won't cause regressions, so I'll do some looking to see if I can prove that to myself. Otherwise, everything here makes sense to me.

Reply:
If it helps, this is the structure from the queue(2) man page:

```
The kevent structure is defined as:

struct kevent {
        uintptr_t  ident;       /* identifier for this event */
        short     filter;       /* filter for event */
        u_short   flags;        /* action flags for kqueue */
        u_int     fflags;       /* filter flag value */
        int64_t   data;         /* filter data value */
        void      *udata;       /* opaque user data identifier */
        uint64_t  ext[4];       /* extensions */
};
```

So the last member of the struct is an array of 4x uint64_t unsigned 64bit integers. Actually, just noticed one of the other struct members has changed as well...